File: backends/ktor/src/main/kotlin/com/hellotimepro/ktor/AppComponents.kt
package com.hellotimepro.ktor

import com.hellotimepro.ktor.config.AppConfig
import com.hellotimepro.ktor.repository.CapsuleRepository
import com.hellotimepro.ktor.repository.FavoriteRepository
import com.hellotimepro.ktor.repository.RefreshTokenRepository
import com.hellotimepro.ktor.repository.UserRepository
import com.hellotimepro.ktor.service.AuthService
import com.hellotimepro.ktor.service.AvatarService
import com.hellotimepro.ktor.service.CapsuleRecommendationService
import com.hellotimepro.ktor.service.CapsuleService
import com.hellotimepro.ktor.service.CapsuleSuggestionService
import com.hellotimepro.ktor.service.FavoriteService
import com.hellotimepro.ktor.service.LlmClient
import com.hellotimepro.ktor.service.MapperService
import com.hellotimepro.ktor.service.PlazaService
import com.hellotimepro.ktor.service.SecurityService
import com.hellotimepro.ktor.service.UserService
import com.hellotimepro.ktor.web.AuthContext

/** 手动依赖装配（Ktor 惯用的简单做法，避免引入 DI 容器）。单例、无状态共享。 */
class AppComponents(val config: AppConfig) {
    val startTime: Long = System.currentTimeMillis()

    // infrastructure / repositories
    private val userRepository = UserRepository()
    private val capsuleRepository = CapsuleRepository()
    private val favoriteRepository = FavoriteRepository()
    private val refreshTokenRepository = RefreshTokenRepository()

    // domain services
    val avatarService = AvatarService(config)
    private val securityService = SecurityService(config)
    private val mapperService = MapperService()
    private val llmClient = LlmClient(config)

    val authService = AuthService(
        config, userRepository, refreshTokenRepository, securityService, mapperService, avatarService,
    )
    val userService = UserService(userRepository, avatarService, mapperService)
    val capsuleService = CapsuleService(capsuleRepository, userRepository, favoriteRepository, mapperService)
    val plazaService = PlazaService(capsuleRepository, userRepository, favoriteRepository, mapperService)
    val favoriteService = FavoriteService(capsuleRepository, favoriteRepository)
    val suggestionService = CapsuleSuggestionService(config, llmClient)
    val recommendationService = CapsuleRecommendationService(config, llmClient)

    val authContext = AuthContext(securityService, userRepository)
}

File: backends/vapor/server/Sources/App/AppComponents.swift
import Vapor
import Foundation

/// 手工依赖装配（Vapor 无 DI 容器，构造函数注入即可）。
struct AppComponents: Sendable {
    let config: AppConfig
    let db: AppDatabase
    let startTime: Date

    let security: SecurityService
    let mapper: MapperService
    let avatars: AvatarService
    let authContext: AuthContext
    let authService: AuthService
    let userService: UserService
    let capsuleService: CapsuleService
    let plazaService: PlazaService
    let favoriteService: FavoriteService
    let suggestionService: SuggestionService
    let recommendationService: RecommendationService

    init(app: Application, config: AppConfig, db: AppDatabase) throws {
        self.config = config
        self.db = db
        self.startTime = Date()

        let users = UserRepository(dbx: db)
        let capsules = CapsuleRepository(dbx: db)
        let favorites = FavoriteRepository(dbx: db)
        let refreshTokens = RefreshTokenRepository(dbx: db)

        let security = SecurityService(config: config)
        let mapper = MapperService()
        let avatars = try AvatarService(config: config)
        let llm = LlmClient(config: config.llm, client: app.client,
                            logger: Logger(label: "app.llm"))

        self.security = security
        self.mapper = mapper
        self.avatars = avatars
        self.authContext = AuthContext(security: security, db: db, users: users)
        self.authService = AuthService(
            config: config, db: db, users: users, refreshTokens: refreshTokens,
            security: security, mapper: mapper, avatars: avatars,
            rateLimiter: LoginRateLimiter(limit: config.loginRateLimitPerMinute)
        )
        self.userService = UserService(db: db, users: users, mapper: mapper, avatars: avatars)
        self.capsuleService = CapsuleService(db: db, capsules: capsules, favorites: favorites, mapper: mapper)
        self.plazaService = PlazaService(db: db, capsules: capsules, mapper: mapper)
        self.favoriteService = FavoriteService(db: db, capsules: capsules, favorites: favorites)
        self.suggestionService = SuggestionService(config: config, llm: llm,
                                                   logger: Logger(label: "app.suggestion"))
        self.recommendationService = RecommendationService(config: config, llm: llm,
                                                           logger: Logger(label: "app.recommendation"))
    }
}
